aptos-move/block-executor/src/executor.rs

```diff
@@ -883,6 +883,12 @@ where
         }
 
         let num_txns = signature_verified_block.len();
+        let concurrency_level = self.config.local.concurrency_level;
```

Comment:
I'd use concurrency_level variable still, and do it via min(), max
since you don't want to use <= 1 ever, you could do

let concurrency_level = max(min(self.config.local.concurrency_level, num_txns / 2), 2) or smt like that.